nvtabular/workflow/workflow.py

```diff
@@ -461,6 +475,9 @@ def _transform_partition(root_df, workflow_nodes, additional_columns=None):
             unseen_columns = set(node.input_columns.names) - seen_columns
             addl_input_cols = addl_input_cols.union(unseen_columns)
 
+            # TODO: Find a better way to remove dupes
+            addl_input_cols = addl_input_cols - set(input_df.columns)
```

Comment:
We should look at this with fresh eyes and either figure out why overlapping columns are happening or find a better fix

Reply:
Here your finding the columns that come from root df right? I guess we should be getting this information from the input_schema for the node op, instead of the input_df, is that way you mean?

Reply:
I'm not even sure what I mean TBH. I just remember writing this comment when you pointed out this was a little sketchy. 😅